code/modules/antagonists/changeling/powers/absorb.dm

```diff
@@ -59,11 +59,31 @@
 	if(user.nutrition < NUTRITION_LEVEL_WELL_FED)
 		user.set_nutrition(min((user.nutrition + target.nutrition), NUTRITION_LEVEL_WELL_FED))
 
-	if(target.mind)//if the victim has got a mind
+	if(target.mind && user.mind)//if the victim and user have minds
 		// Absorb a lizard, speak Draconic.
 		user.copy_known_languages_from(target)
 
-		target.mind.show_memory(user, 0) //I can read your mind, kekeke. Output all their notes.
+		var/datum/mind/suckedbrain = target.mind
+		user.mind.memory += "<BR><b>We've absorbed [target]'s memories into our own...</b><BR>[suckedbrain.memory]<BR>"
+		if(LAZYLEN(suckedbrain.antag_datums))//this entire massive block is for antag memories and objectives
+			for(var/datum/antagonist/antag_types in suckedbrain.antag_datums)
+				var/list/all_objectives = list()
+				all_objectives |= antag_types.objectives
+				if(antag_types.antag_memory)
+					user.mind.memory += "[antag_types.antag_memory]<BR>"
+				if(LAZYLEN(all_objectives))
+					user.mind.memory += "<B>Objectives:</B>"
+					var/obj_count = 1
+					for(var/datum/objective/objective in all_objectives)
+						user.mind.memory += "<br><B>Objective #[obj_count++]</B>: [objective.explanation_text]"
+						var/list/datum/mind/other_owners = objective.get_owners() - suckedbrain
+						if(other_owners.len)
+							user.mind.memory += "<ul>"
+							for(var/datum/mind/M in other_owners)

```

Comment:
```suggestion
							for(var/mind in other_owners)
								var/datum/mind/M = mind
```

Reply:
Just curious why this method is better. Will implement it, but I had assumed that asking for a specific path in the for() would only return that path, so functionality shouldn't change?